modules/src/ics03_connection/connection.rs

```diff
@@ -176,3 +165,103 @@ pub fn validate_version(version: String) -> Result<String, String> {
     }
     Ok(version)
 }
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum State {
+    Uninitialized = 0,
+    Init,
+    TryOpen,
+    Open,
+}
+
+impl State {
+    /// Yields the State as a string.
+    pub fn as_string(&self) -> &'static str {
+        match self {
+            Self::Uninitialized => "UNINITIALIZED",
+            Self::Init => "INIT",
+            Self::TryOpen => "TRYOPEN",
+            Self::Open => "OPEN",
+        }
+    }
+
+    /// Parses the State from a i32.
+    pub fn from_i32(nr: i32) -> Self {
+        match nr {
+            1 => Self::Init,
+            2 => Self::TryOpen,
+            3 => Self::Open,
+            _ => Self::Uninitialized,
+        }
+    }
+}
+
+impl std::str::FromStr for State {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
```

Comment:
Should we expect that when calling this function the words will be always uppercase ? Will there be cases where a value like `Init` or 'init` could be passed ? If so, an case-sensitive is not important maybe have `match s.to_uppercase()` and only fail if the keyword for state passed is something totally unknown e.g. `CLOSED`

Reply:
Actually I believe that we don't need the `from_str()` functions, they are currently used from tests only. Will double check and remove from both connection and channel.